### PR TITLE
Fix wrong dialog position when it is placed above its target.

### PR DESCRIPTION
**Pull request by shaoshing**
This PR fixes issue #3325.

Before:

screen shot 2015-12-09 at 11 45 00 am

After:

screen shot 2015-12-09 at 11 45 13 am

### CODE REVIEW

File: src/js/jsx/shared/Dialog.jsx [hunk 1]
```diff
@@ -211,34 +211,40 @@ define(function (require, exports, module) {
...
-                            placedDialogTop = clientHeight -
+                            topPosition = clientHeight -
                                 dialogBounds.height - dialogMarginTop - dialogMarginBottom;
                         }
                     }
 
                     this.setState({
-                        "topPosition": placedDialogTop + "px",
+                        "topPosition": topPosition + "px",
+                        "bottomPosition": bottomPosition + "px",
```
Comment (mcilroyc) [inline src/js/jsx/shared/Dialog.jsx]: I can't find any user-facing issue from this... but it is possible that `topPosition` and `bottomPosition` to be accessed without being defined... resulting in the string "undefinedpx" being added to state.
ss

**Comment by mcilroyc** [comment]
works fine AFAICT.  But I have one gripe that is just legitimate enough to push back to @shaoshing

**Comment by shaoshing** [comment]
@mcilroyc Thanks for the careful review, that's a really good catch. I've fix the `undefinedpx` value. I also made some extra change to move some functions in `Dialog.jsx` to their proper positon. Back to you.

**Comment by mcilroyc** [comment]
bueno

**Pull request merged**